### DISCUSSION TITLE
Copilot in Windows + WSL + VSCode: sometimes doesn't work, sometimes works fine, I don't get it.

### DISCUSSION THREAD

**Discussion by JoannaFalkowska** [community/community#29269] [Copilot Conversations]
It worked fine for months, but now it suddenly stopped.

When I open a VSCode window in "vanilla" Windows, Copilot works completely fine, i.e. I can create a new file and it starts giving me suggestions as expected.

However when I open a VSCode window in WSL, Copilot extension manages to start up and authenticate, but then it gets stuck on "Synthesizing 0/10 solutions". I can see no errors in the console. It's just stuck.

I tried:
* removing password to the default keyring in WSL (using an utility called `seahorse`)
* deleting the entire default keyring in WSL (again, with `seahorse`) and re-authenticating Github
* reinstalling Copilot VSCode extension
* restarting the WSL machine

None of those helped, and I'm completely lost as to why Copilot is suddenly unable to cooperate with my WSL. I don't use any VPN, and haven't changed any settings this past week.

I'd appreciate any kind of pointers as to how I could debug this.

**Comment by davecheney** [comment]
@JoannaFalkowska I'm sorry you're having difficulties. Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by JoannaFalkowska** [reply]
Sure. The entire log looks pretty much like this:

```
[INFO] [default] [2022-08-12T11:56:10.135Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:56:15.203Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:56:15.384Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:56:15.916Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:56:16.113Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:56:16.859Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:56:17.044Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:57:43.842Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:57:45.239Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:57:51.079Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [ghostText] [2022-08-12T11:58:09.382Z] Cancelled during debounce
[INFO] [default] [2022-08-12T11:58:21.594Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:58:21.948Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [ghostText] [2022-08-12T11:58:22.394Z] Cancelled during debounce
[INFO] [default] [2022-08-12T11:58:22.492Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [default] [2022-08-12T11:59:39.005Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [ghostText] [2022-08-12T14:50:36.776Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-08-12T14:50:36.777Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-08-12T14:50:36.777Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-08-12T14:50:36.781Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-08-12T14:50:36.912Z] Cancelled during debounce
[INFO] [default] [2022-08-12T14:50:37.063Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus
[INFO] [solutions] [2022-08-12T14:51:02.153Z] prompt: {"prefix":"// Language: javascript\nhelloWorld() {\n","suffix":"","isFimEnabled":false,"promptElementRanges":[{"kind":"LanguageMarker","start":0,"end":24},{"kind":"BeforeCursor","start":24,"end":41}]}
[INFO] [default] [2022-08-12T14:51:02.154Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-d-centralus

```

**Reply by davecheney** [reply]
Copilot is working, however there are some words in your source code that we cannot offer a completion for. In that case, copilot will not return a completion.

**Reply by JoannaFalkowska** [reply]
@davecheney I'm sorry, I don't think that's possible. Whatever I type, it _never_ returns _any_ completions whatsoever - but **only in WSL**. In Windows I can get plenty of completions all the time.

In the example logged above, as you can see, I typed out an empty `helloWorld() { }` function and waited for a completion inside the brackets. I never got any.

**Reply by davecheney** [reply]
@JoannaFalkowska thank you for correcting me. I've passed this thread onto the engineers who are managing the experiment -d endpoint. I suggest restarting vscode to try and shake off the assignment to that experiment endpoint and hopefully get copilot working again.

**Reply by davecheney** [reply]
@JoannaFalkowska I've disabled the experiment, hopefully this should resolve itself shortly.

**Reply by JoannaFalkowska** [reply]
Hey @davecheney, thank you for your response.

2 days later, after restarting VSCode etc., it seems that I'm now assigned to `experiment-e` instead of `experiment-d`. It still doesn't return anything at all in WSL, but **the same endpoint** does correctly return suggestions in Windows. So I don't think it's a problem with your experiment itself. 

It's probably some internal networking or auth-related issue on my side, but I honestly don't know how to debug it. Everything else works generally fine, and the extension itself seems to be thinking it's authenticated correctly.

(This is likely unrelated, but I recently had some problems accessing `git://` urls when running a simple `npm install`, and had to [switch the protocol in git global config](https://gist.github.com/taoyuan/bfa3ff87e4b5611b5cbe) to `https://`. I changed it just yesterday though, so after I already was having problems with Copilot. Do you think it could be any kind of clue?)

---

As for logs:

WSL VSCode :
```
[INFO] [solutions] [2022-08-16T13:29:38.447Z] prompt: {"prefix":"// Language: javascript\nhelloWorld() {\n","suffix":"","isFimEnabled":false,"promptElementRanges":[{"kind":"LanguageMarker","start":0,"end":24},{"kind":"BeforeCursor","start":24,"end":41}]}
[INFO] [default] [2022-08-16T13:29:38.510Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-e-centralus
[INFO] [default] [2022-08-16T13:30:17.243Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-e-centralus

(...zero incoming responses even after waiting for >90s)
```

Windows VSCode :
```
[INFO] [solutions] [2022-08-16T13:42:48.460Z] prompt: {"prefix":"// Language: javascript\nhelloWorld() {\n","suffix":"","isFimEnabled":false,"promptElementRanges":[{"kind":"LanguageMarker","start":0,"end":24},{"kind":"BeforeCursor","start":24,"end":41}]}
[INFO] [default] [2022-08-16T13:42:48.518Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-e-centralus
[INFO] [default] [2022-08-16T13:42:49.191Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-e-centralus/completions] took 673 ms
[ERROR] [streamChoices] [2022-08-16T13:42:58.875Z] Error parsing JSON stream data,{"error":{"message":"request timeout"}}
[INFO] [streamChoices] [2022-08-16T13:42:58.876Z] solution 0 returned. finish reason: [Iteration Done] finishOffset: [undefined] completionId: [{cmpl-5fr8CjNBnDaMOLyxSuuMW1zwNfia2}] created: [{1660657368}]
[INFO] [solutions] [2022-08-16T13:42:58.880Z] Open Copilot completion: [  console.log("Hello World!");
}

// Language: python
def helloWorld():
  print("Hello World!")

// Language: java
public void helloWorld() {
  System.out.println("Hello World!");
}

// Language: c++
void helloWorld() {
  std::cout << "Hello World!";
}

(...more solutions in similar vein)
```

I also tried to use the "Github Copilot Nightly" VSCode extension in WSL instead, and it used a different endpoint (`codex`), but that one also never returned anything:

```
[INFO] [solutions] [2022-08-16T13:37:33.176Z] prompt: {"prefix":"// Language: javascript\nhelloWorld() {\n","suffix":"","isFimEnabled":false,"promptElementRanges":[{"kind":"LanguageMarker","start":0,"end":24},{"kind":"BeforeCursor","start":24,"end":41}]}
[INFO] [default] [2022-08-16T13:37:45.718Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2022-08-16T13:37:45.721Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-08-16T13:37:45.722Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-08-16T13:37:45.722Z] Cancelled after requestMultiline
[INFO] [default] [2022-08-16T13:37:46.016Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-08-16T13:37:59.616Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2022-08-16T13:38:35.332Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

---

Please let me know if you have any other ideas as to how I could debug this, I appreciate your help.

**Reply by JoannaFalkowska** [reply]
Hello @davecheney, sorry for nudging you, but since my Copilot trial has ended I'm now not sure if I should be purchasing the subscription or not. I I can't get it working in WSL, then it's no point for me. But maybe I can try to debug it even without the subscription being paid for, somehow? (Or, is it likely that I would get refunded if it didn't work?) I'm quite determined to get it up to speed again, I had a great experience with it.

I can see this thread got upvoted by a few people, so possibly it's not just me having this problem. Have you maybe heard about any new ideas why would Copilot not work in WSL in the meantime, since your last message?

**Comment by JoannaFalkowska** [comment]
This is STILL unresolved, I'm STILL struggling with the same problem, without any idea why does it happen.

On good days, I open my WSL files in VSCode and Copilot instantly works like a charm, swiftly returning completions all day.

On bad days, in exactly the same setting, it returns no completions, or manages to do so only for isolated very simple files, while completely blanking and timing out when asked for completions inside a complicated TS repo.

It's frankly ridiculous, completely unreliable, deciding to work or not work without any rhyme or reason. I'd like to understand at least _why_ and whether I can do _anything_ to help it function properly. Because as it is now, it's a paid feature working in a way that can hardly be considered out of beta.

**Reply by Markkos89** [reply]
Same here... totally randomly works or not.... I don't get it

**Reply by johnashu** [reply]
Another year and I am here with the same issues lol.

**Reply by Kagura-Ahad** [reply]
bruh

this is my output.
[ERROR] [default] [2023-11-26T20:56:28.487Z] Extension activation failed: "connect ETIMEDOUT 20.207.73.85:443"

what do i do??

**Reply by techybolek** [reply]
All of a sudden I'm hacing the same issue. Works on Windows, not on WSL2.  HAs anyone been able to resolve this issue?

**Reply by tetthys** [reply]
Still unresolved...

**Reply by fmPeretti** [reply]
Still having the same behaviour... Sometimes its works other times it just dont want to suggest anything

**Reply by ritiek** [reply]
@Kagura-Ahad Hi. I'm guessing this might be helpful in your case (DNS resolves to same bad IP):
https://github.com/orgs/community/discussions/29269#discussioncomment-10094288

**Reply by Tarisper** [reply]
> @Kagura-Ahad Hi. I'm guessing this might be helpful in your case (DNS resolves to same bad IP): https://github.com/orgs/community/discussions/29269#discussioncomment-10094288

This method really works.
Need to go to the settings.json add:
```json
"remote.extensionKind": {
        "GitHub.copilot": [ "ui" ],
        "GitHub.copilot-chat": ["ui"]
}
```

**Comment by Min-Ho-Lim** [comment]
![image](https://user-images.githubusercontent.com/13793276/205128218-f1c36680-4483-41ec-9157-679ddb26175b.png)


This happens to me as well. I recently formatted my pc and installed the latest windows 10 pro.
I installed `Ubuntu` with WSL2 to use it as my dev setup alongwith VSCode.
Also, I installed Copilot as well. It had been working fine last few days, but suddenly, from today, it stopped working.

```
[INFO] [default] [2022-12-01T18:01:32.028Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```
The Copilot log does not show any error as @JoannaFalkowska mentioned when I try to get any suggestion from Copilot.

```
[ERROR] [default] [2022-12-01T18:07:30.787Z] Request Error: connect ETIMEDOUT 20.80.64.28:443
```
But, If I waited long enough, it does return with the timout error.

This is the list that I have tried
1. Reinstall Copilot extension
2. Disabled Copilot and enabled it back
3. Signed out and signed in my Github account from VSCode
4. Turned off all VPN connections
5. Used Google public DNS
6. Replaced the nameserver to the Google public DNS in `~/etc/resolv.conf`
7. Changed WSL2 to WSL1

I am not sure what is going on.

I will update the post if I figure out the solution.

**Comment by Min-Ho-Lim** [comment]
Alright, it has been almost a month since I posted. I do not have the exact solution, but I kinda know what is the problem.

So, Copilot does use multiple servers to get responses, but the problem is once Copilot connected to a server, it does not get changed even if the connection gets timed out.

I am not sure why Copilot does not automatically change the connection server. 

But, I think sometimes the copilot server thinks some connection from "WSL" is **evil**, so they block the IP address for a short period time. This is why we experience such cases.

The workaround is using a VPN. 

Since the communication between your IP and the Copilot server is not established, you need to changed your IP address.

**Comment by Michael-Bidencope** [comment]
Hey,
I also I had the same problem with WSL2,  VSCode  , and Copilot, I was not able to get a VPN to work as it was just causing more problems. However, I was able to use VSCode's proxy and Squid for Windows(proxy) in order to fix this problem. [Here ](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-visual-studio-code#configuring-proxy-settings-for-github-copilot) is the github post to configuring proxy settings for GitHub Copilot.

**Reply by hervebru** [reply]
Thanks @MichaelBidencopeCP, that's working with a local proxy running on Windows.

Nevertheless I hope Github is going to fix this because that remains an annoyance.

In case Github is still reading us (@davecheney ?), the issue in this thread is NOT about having no connectivity at all without proxy:
the vscode github extention running in the WSL context do have connectivity to github. From a WSL terminal, the _ping test
to  githubusercontent is always successful without proxy:

```
(base) user@WSL2:~$ curl https://copilot-proxy.githubusercontent.com/_ping
{"now":1676046306,"status":"ok"}
```

However, without proxy, the extension fails to return anything most of the time:

```[INFO] [default] [2023-02-10T12:34:10.878Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-02-10T12:34:11.043Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [ghostText] [2023-02-10T12:34:11.257Z] Cancelled during debounce
[INFO] [default] [2023-02-10T12:34:11.347Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-02-10T12:34:11.571Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[INFO] [default] [2023-02-10T12:34:12.645Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

**Comment by techybolek** [comment]
> In case Github is still reading us (@davecheney ?), the issue in this thread is NOT about having no connectivity at all without proxy: the vscode github extention running in the WSL context do have connectivity to github. From a WSL terminal, the _ping test to githubusercontent is always successful without proxy:
> 
> ```
> (base) user@WSL2:~$ curl https://copilot-proxy.githubusercontent.com/_ping
> {"now":1676046306,"status":"ok"}
> ```
> 
Interesting. My issue may be slightly different. The curl won't return anything on WSL2, while it returns OK on windows. Any suggestions?

**Comment by robharding** [comment]
Here's how I set up squid proxy with docker for a workaround:

squid.conf
```
http_port 3128
http_access allow all
cache deny all
forwarded_for delete
via off

# Enable SSL bump
ssl_bump allow all

# Don't verify server certificates
sslproxy_cert_error allow all
tls_outgoing_options flags=DONT_VERIFY_PEER
```

run.sh
```
#!/bin/bash

CONTAINER_NAME="proxy-server-container"
IMAGE_NAME="proxy-server"
PORT=3128

# Check if the container is running
if docker ps --format "{{.Names}}" | grep -Eq "^${CONTAINER_NAME}\$"; then
    echo "Stopping running container..."
    docker stop $CONTAINER_NAME
    docker rm $CONTAINER_NAME
elif docker ps -a --format "{{.Names}}" | grep -Eq "^${CONTAINER_NAME}\$"; then
    echo "Removing stopped container..."
    docker rm $CONTAINER_NAME
fi

# Start the container
echo "Starting the container..."
docker run -d --name proxy-server-container -p $PORT:$PORT proxy-server

echo "Done."
```

Dockerfile
```
# Use the official Ubuntu 20.04 image as the base
FROM ubuntu:20.04

# Set environment variables to avoid interactive installation
ENV DEBIAN_FRONTEND=noninteractive

# Update package list and install Squid
RUN apt-get update && \
    apt-get install -y squid

# Copy custom Squid configuration file
COPY squid.conf /etc/squid/squid.conf

# Expose the Squid port
EXPOSE 3128

# Run Squid in the foreground
CMD ["squid", "-N"]
```

build.sh
```
#!/bin/bash

docker build -t proxy-server .
```

then set http://0.0.0.0:3128 as your proxy in vscode.

NOTE: I have to remove the proxy when installing/updating extentions

**Comment by tetthys** [comment]
This shitty issue still hasn't been resolved lol

**Reply by tetthys** [reply]
This issue also appears quite often when using paid VPN services.

At least this method seems to work for me.

https://github.com/tetthys/copilotshittyissue-wsl2vpnproxy

Try this repo.

**Reply by bauhaas** [reply]
worked like a charm !  Tried the most recent solution from @fredpiuma, sadly, not working in my case

**Comment by guaregua19** [comment]
Same issue here

**Comment by ClaraSchumann** [comment]
same

**Comment by niladri-1** [comment]
1. **squid.conf**:

   ```conf
   # Set up Squid proxy configuration
   http_port 3128
   http_access allow all
   cache deny all
   forwarded_for delete
   via off

   # Enable SSL bump
   ssl_bump allow all

   # Disable server certificate verification
   sslproxy_cert_error allow all
   tls_outgoing_options flags=DONT_VERIFY_PEER
   ```

2. **run.sh**:

   ```bash
   #!/bin/bash

   # Define container and image names
   CONTAINER_NAME="proxy-server-container"
   IMAGE_NAME="proxy-server"
   PORT=3128

   # Check if the container is already running or exists
   if docker ps --format "{{.Names}}" | grep -Eq "^${CONTAINER_NAME}\$"; then
       echo "Stopping running container..."
       docker stop $CONTAINER_NAME
       docker rm $CONTAINER_NAME
   elif docker ps -a --format "{{.Names}}" | grep -Eq "^${CONTAINER_NAME}\$"; then
       echo "Removing stopped container..."
       docker rm $CONTAINER_NAME
   fi

   # Start the container
   echo "Starting the container..."
   docker run -d --name proxy-server-container -p $PORT:$PORT proxy-server

   echo "Done."
   ```

3. **Dockerfile**:

   ```Dockerfile
   # Use the official Ubuntu 20.04 image as the base
   FROM ubuntu:20.04

   # Set environment variables to avoid interactive installation
   ENV DEBIAN_FRONTEND=noninteractive

   # Update package list and install Squid
   RUN apt-get update && \
       apt-get install -y squid

   # Copy custom Squid configuration file
   COPY squid.conf /etc/squid/squid.conf

   # Expose the Squid port
   EXPOSE 3128

   # Run Squid in the foreground
   CMD ["squid", "-N"]
   ```

4. **build.sh**:

   ```bash
   #!/bin/bash

   # Build the Docker image
   docker build -t proxy-server .
   ```

**Reply by tetthys** [reply]
I was inspired by this. Thank you!

https://github.com/tetthys/copilotshittyissue-wsl2vpnproxy

**Comment by PsikxCat** [comment]
Same issue

**Comment by fredpiuma** [comment]
[SOLVED] Downgrade to version v1.188.0

Problem: There is an issue in the latest version of Github Copilot that causes it to work on Windows but not WSL.

**Reply by INGCapaDev** [reply]
Thanks, that's working in my case 👍

**Reply by binbalenci** [reply]
This works in my case too. Thanks!

**Reply by tetthys** [reply]
The bug the copilot icon disappears in wsl has been resolved from v1.189.0.

**Comment by NehalDamania** [comment]
I too am facing this issue.

**Comment by pbotas** [comment]
Same here... cancelling subscription. What's the best alternative?

**Reply by tetthys** [reply]
Have you ever tried [this repo](https://github.com/tetthys/copilotshittyissue-wsl2vpnproxy)?

**Reply by pbotas** [reply]
no, not interested on these kinds of workarounds

**Reply by tetthys** [reply]
You need to set up a proxy. I don't know the exact reason, but the problem was solved after setting up squid proxy. That repo just sets up spuid proxy with docker-compose.

**Comment by milliephanillie** [comment]
Same issue

**Comment by jahanzaibajmal** [comment]
Same, not working in wsl2 but working fine in windows. If you disable and then enable it run 1 or 2min then again not work

**Comment by NehalDamania** [comment]
Recently, I was asked to try VSCode Insiders build. And it is working on VSCode insiders build on WSL2. Normal VSCode build it is not working

**Comment by ritiek** [comment]
I had this issue where half the times I wasn't able to clone GitHub repositories from inside of a docker container, it would get stuck. Even so, it cloned fine if I cancelled the clone operation and re-tried cloning.

I also found out that curl had a similar issue with api.github.com where half the times it'd get stuck while it worked fine the other half:
```bash
$ curl -v --trace-time https://api.github.com
15:36:50.430458 * Host api.github.com:443 was resolved.
15:36:50.430716 * IPv6: (none)
15:36:50.430768 * IPv4: 20.207.73.85
15:36:50.430800 *   Trying 20.207.73.85:443...
15:36:50.437772 * Connected to api.github.com (20.207.73.85) port 443
15:36:50.440882 * ALPN: curl offers h2,http/1.1
15:36:50.441273 * TLSv1.3 (OUT), TLS handshake, Client hello (1):
15:36:50.465983 *  CAfile: /etc/ssl/certs/ca-certificates.crt
15:36:50.466042 *  CApath: /etc/ssl/certs

(stuck here 3 mins)

15:39:01.414365 * Recv failure: Connection reset by peer
15:39:01.414422 * OpenSSL SSL_connect: Connection reset by peer in connection to api.github.com:443
15:39:01.414450 * Closing connection
curl: (35) Recv failure: Connection reset by peer
```

I tried to debug this. In my case, it looks that when trying to DNS resolve api.github.com (`dig api.github.com`), it would return either of these IP addresses: 140.82.121.6 or 20.207.73.85. So, I hardcoded these resolved IP in my `/etc/hosts` one at a time, like so:
```
# /etc/hosts
# ...
140.82.121.6   api.github.com
```
and then tried to curl/git clone and everything worked fine. It turns out when I hardcoded the resolution to 20.207.73.85, I was able to reproduce the problem of things getting stuck consistently.

I'm guessing this should have a bit to do with GitHub side of server things. Although, I'm still not sure why it doesn't get stuck even with the bad IP when I git clone/curl on my host OS (unlike inside of a docker container).

**Reply by sameer55chauhan** [reply]
Worked for me, thanks! I performed the same for api.githubcopilot.com as well.

**Comment by panhypersebastos** [comment]
Same issue here. Copilot working fine under Windows, but erratic behavior under WSL...

**Comment by Jnn002** [comment]
Same here, but I just go to connections > manage trusted extensions > manage account preferences > use a new account and select my github account again .... unfortunately this is not a definitive solution and you will have to do it often, some people said this is a “temporary bug”, I hope this won't last long....

**Comment by Ciantic** [comment]
I have the same issue. 

This shouldn't be discussion but a bug.

Sometimes nothing seems to work, then suddenly it restarts working.

I've noticed that it starts to work again (sometimes) when clicking the model change button and switching it to something else and then back:

<img src="https://github.com/user-attachments/assets/fba3a4a7-0d51-46a5-9f81-650db027e201" width="300" />

Sometimes, I start getting errors like:

![image](https://github.com/user-attachments/assets/ff26ea08-9962-45d8-b37d-76c12788d0cc)

As if the whole extension was not working.

**Comment by Ali-sbp** [comment]
same issue here, copilot works perfectly fine on vanilla vs code but on WSL it doesn't authenticate properly, nothing works

**Comment by jvargh7** [comment]
Doesn't work for me as well. 

But found this workaround on [stackoverflow](https://stackoverflow.com/questions/78719710/copilot-is-not-working-is-wsl-remote-connection).

**Reply by mansenfranzen** [reply]
> Doesn't work for me as well.
> 
> But found this workaround on [stackoverflow](https://stackoverflow.com/questions/78719710/copilot-is-not-working-is-wsl-remote-connection).

I can confirm that adding this actually worked for me:

```json
"remote.extensionKind": {
        "GitHub.copilot": [ "ui" ],
        "GitHub.copilot-chat": ["ui"]
}
```

**Reply by dheerajawale-svg** [reply]
Tried this, no luck

**Comment by zkw13** [comment]
2025-06-19 12:20:35.402 [info] [certificates] Removed 12 expired certificates
2025-06-19 12:20:36.571 [info] [fetcher] Using Helix fetcher.
2025-06-19 12:20:37.418 [info] [fetcher] Using Helix fetcher.

randomly not working.

**Comment by dheerajawale-svg** [comment]
<img width="485" height="525" alt="image" src="https://github.com/user-attachments/assets/340e0c08-628c-4efd-b389-78a68aee730d" />

Agent mode cannot even modify files when its in WSL mode

**Comment by aiw3ndil** [comment]
It works with this settings.json
```
{
    "github.copilot.advanced": {
        "debug.useNodeFetcher": true,
        "debug.useNodeFetchFetcher": false
    },
    "remote.extensionKind": {
    "github.copilot": "ui",
    "github.copilot-chat": "ui"
    },
    "files.watcherExclude": {
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true
    },
    "files.useExperimentalFileWatcher": false,
    "remote.WSL.fileWatcher.polling": true,
    "http.proxy": "http://172.28.32.1:8001", // ip route | grep default
    "http.proxySupport": "on",
    "http.proxyStrictSSL": false,
    "github.copilot.advanced.debug.useElectronFetcher": false,
    "github.copilot.advanced.debug.useNodeFetcher": true, 
    "github.copilot.advanced.debug.useNodeFetchFetcher": false
}
```

I'm trying the trial now and everything seems to be working perfectly.

**Reply by elminster-aom** [reply]
This doesn't help in my case.

When the extensions runs locally on Windows (see [discussioncomment](https://github.com/orgs/community/discussions/29269#discussioncomment-13325238) above), the Copilot works but it messes up between the windows directory separator (`\`) and the WSL-Linux one (`/`). 

After replicating your setup, I still get this same error; Asking Copilot to create `\home\johndoe\src\cmd\hello\main.go` but responds:
> The model wants to edit files outside of your workspace (`\home\johndoe\src\cmd\hello\main.go`).
